### PR TITLE
Migration to terraform 0.12

### PR DESCRIPTION
**Pull request by v47**
Migration to terraform 0.12.

### CODE REVIEW

**Comment by v47** [comment]
Practicaly i just needed to replace `"${expression}"` with `expression`.
> Terraform 0.11 and earlier required all non-constant expressions to be
> provided via interpolation syntax, but this pattern is now deprecated. To
> silence this warning, remove the "${ sequence from the start and the }"
> sequence from the end of this expression, leaving just the inner expression.
> 
> Template interpolation syntax is still used to construct strings from
> expressions when the template includes multiple interpolation sequences or a
> mixture of literal strings and interpolations. This deprecation applies only
> to templates that consist entirely of a single interpolation sequence.

File: main.tf [hunk 1]
```diff
@@ -4,26 +4,26 @@ data "aws_route53_zone" "default" {
...
-  domain_name               = "${var.domain_name}"
+  domain_name               = var.domain_name
   validation_method         = "DNS"
-  subject_alternative_names = "${concat(var.alternative_domain_names, list(format("*.%s", var.domain_name)))}"
+  subject_alternative_names = concat(var.alternative_domain_names, list(format("*.%s", var.domain_name)))
 
-  tags = "${merge(
+  tags = merge(
     map(
       "Name", "${var.name}"
```
Comment (krzywiecki) [inline main.tf]: @v47 Don't you need to replace this as well?

File: main.tf [hunk 2]
```diff
@@ -4,26 +4,26 @@ data "aws_route53_zone" "default" {
...
 }
 
 resource "aws_route53_record" "default" {
-  name    = "${aws_acm_certificate.default.domain_validation_options.0.resource_record_name}"
-  type    = "${aws_acm_certificate.default.domain_validation_options.0.resource_record_type}"
-  zone_id = "${data.aws_route53_zone.default.id}"
+  name    = aws_acm_certificate.default.domain_validation_options.0.resource_record_name
+  type    = aws_acm_certificate.default.domain_validation_options.0.resource_record_type
+  zone_id = data.aws_route53_zone.default.id
   records = ["${aws_acm_certificate.default.domain_validation_options.0.resource_record_value}"]
```
Comment (krzywiecki) [inline main.tf]: And this one?

File: main.tf [hunk 3]
```diff
@@ -4,26 +4,26 @@ data "aws_route53_zone" "default" {
...
   ttl     = 60
 }
 
 resource "aws_acm_certificate_validation" "default" {
-  certificate_arn         = "${aws_acm_certificate.default.arn}"
+  certificate_arn         = aws_acm_certificate.default.arn
   validation_record_fqdns = ["${aws_route53_record.default.fqdn}"]
```
Comment (krzywiecki) [inline main.tf]: And this?

File: main.tf (hunk 1 above)
Comment (v47) [reply main.tf]: Here I need to use it as a string.

**Comment by v47** [comment]
Both are strings.

**Pull request merged**